File: mpp/Lab9/javaPb9/services/src/main/java/ro/mpp2024/IService.java
package ro.mpp2024;

import ro.mpp2024.model.Arbitru;
import ro.mpp2024.model.Participant;

import java.util.Map;

public interface IService {
    Arbitru login(String username, String password, IManageObserver client) throws ManageException;

    void addRezultat(Long idParticipant,String nume,String prenume, String idProba, long puncte) ;

    Map<Participant,Long> getParticipantiAlfabetic();

    Map<Participant,Long> getParticipantiPuncteDesc(String Proba);

    Iterable<Participant> getParticipanti();

    void logout(Arbitru user,IManageObserver client) throws ManageException;
}

File: mpp/Lab10/csharpPb9/services/services/IService.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace services.services
{
    public interface IService
    {
        Arbitru login(string username, string password, IManageObserver client);

        void addRezultat(long id, string nume, string prenume, string idProba, long punctaj);

        Dictionary<Participant, long> getParticipantiAlfabtic();
        Dictionary<Participant, long> getParticipantiPuncteDesc(string idProba);

        IEnumerable<Participant> GetParticipants();

        void logutOut(Arbitru arbitru, IManageObserver client);
    }
}